packages/modeling/src/measurements/measureCenterOfMass.js

```diff
@@ -0,0 +1,90 @@
+const flatten = require('../utils/flatten')
+
+const vec3 = require('../maths/vec3')
+
+const geom2 = require('../geometries/geom2')
+const geom3 = require('../geometries/geom3')
+const path2 = require('../geometries/path2')
+
+const measureBoundingBox = require('./measureBoundingBox')
+
+const cacheOfCenterOfMass = new WeakMap()
+
+/*
+ * Measure the center of mass for the given geometry.
+ * The given geometry is infinitely thin, so there's no real mass, just a center.
+ * @return {Array} the center of mass for the geometry
+ */
+const measureCenterOfMassGeom2 = (geometry) => {
+  let centerOfMass = cacheOfCenterOfMass.get(geometry)
+  if (centerOfMass !== undefined) return centerOfMass
+
+  const bounds = measureBoundingBox(geometry)
+  centerOfMass = [0, 0, 0]
+  centerOfMass[0] = (bounds[0][0] + ((bounds[1][0] - bounds[0][0]) / 2))
+  centerOfMass[1] = (bounds[0][1] + ((bounds[1][1] - bounds[0][1]) / 2))
+  centerOfMass[2] = (bounds[0][2] + ((bounds[1][2] - bounds[0][2]) / 2))
+
+  cacheOfCenterOfMass.set(geometry, centerOfMass)
+  return centerOfMass
+}
+
+/*
+ * Measure the center of mass for the given geometry.
```

Comment:
Code looks good. I think the math checks out, but I'm less certain, there. :). I'll see if I can throw some tests at it, too.

Reply:
See the centroid.pdf for the math. ;)